recipes/msix/all/patches/1.7/cmake.patch

```diff
@@ -0,0 +1,287 @@
+@@ -103,6 +113,7 @@ configure_file(${MSIX_PROJECT_ROOT}/Package.nuspec.cmakein ${MSIX_BINARY_ROOT}/P
+ configure_file(${MSIX_PROJECT_ROOT}/Microsoft.MSIX.Packaging.targets ${MSIX_BINARY_ROOT}/build/native/${MSIX_NUGET_NAME}.targets)
+ message(STATUS "Package.Nuspec created")
+ message(STATUS "--------------------------------")
++endif()
+ 
+ # Configure license txt
+ configure_file(${MSIX_PROJECT_ROOT}/LICENSE ${MSIX_BINARY_ROOT}/build/LICENSE)
+@@ -170,9 +181,6 @@ if((MACOS) OR (IOS))
+             message(FATAL_ERROR "Unsupported iOS version: ${IOS_DEPLOYMENT_TARGET}, this project requires at least iOS version 10.0")
+         endif()
+     endif()
+-elseif(AOSP OR LINUX)
+-    # Static libraries must be position independent to be linked with a shared object.
+-    set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+ endif()
+ 
+ # Mac needed variables
+@@ -183,10 +191,6 @@ set(CMAKE_MACOSX_RPATH ON)
+ #set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+ #set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+ 
+-add_subdirectory(lib)
+-message(STATUS "libs processed")
+ add_subdirectory(src)
+ message(STATUS "src processed")
+-add_subdirectory(sample)
+-message(STATUS "sample processed")
+ message(STATUS "DONE!")
+diff --git a/cmake/msix_options.cmake b/cmake/msix_options.cmake
+index faba2fbe..5ccdb087 100644
+--- a/cmake/msix_options.cmake
++++ b/cmake/msix_options.cmake
+@@ -13,7 +13,7 @@ option(USE_SHARED_ZLIB "Choose the type of dependency for zlib, Use the -DUSE_SH
+ option(USE_STATIC_MSVC "Windows only. Pass /MT as a compiler flag to use the staic version of the run-time library. Default is 'off' (dynamic)" OFF)
+ option(SKIP_BUNDLES "Removes bundle functionality from the MSIX SDK. Default is 'off'" OFF)
+ option(MSIX_PACK "Include packaging features for the MSIX SDK. Not supported for mobile. Default is 'off'" OFF)
+-option(USE_MSIX_SDK_ZLIB "Use zlib implementation under lib/zlib. If off, uses inbox compression library. For Windows and Linux this is no-opt." OFF)
++option(USE_EXTERNAL_ZLIB "Use an external zlib package. If off, uses inbox compression library. For Windows and Linux this is no-opt." OFF)
+ 
+ set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel. Use the -DCMAKE_BUILD_TYPE=[option] to specify.")
+ set(XML_PARSER "" CACHE STRING "Choose the type of parser, options are: [xerces, msxml6, javaxml].  Use the -DXML_PARSER=[option] to specify.")
+@@ -70,8 +70,8 @@ if(MSIX_PACK)
+     if(AOSP OR IOS)
+         message(FATAL_ERROR "Packaging is not supported for mobile devices.")
+     elseif(MAC)
+-        if(NOT USE_MSIX_SDK_ZLIB)
+-            message(FATAL_ERROR "Using libCompression APIs and packaging features is not supported. Use -DUSE_MSIX_SDK_ZLIB=on")
++        if(NOT USE_EXTERNAL_ZLIB)
++            message(FATAL_ERROR "Using libCompression APIs and packaging features is not supported. Use -DUSE_EXTERNAL_ZLIB=on")
```

Comment:
Renaming names of cmake options in patches is useless.
It's okay to have the name of an option in a conan recipe to be diffferent then the option in the cmake script.
These changes trouble the patch + make it bigger for no reason.

Reply:
[Renaming reverted](https://github.com/conan-io/conan-center-index/pull/6475/commits/e050a90f69494cd24755a65778701df358a08588)